### PR TITLE
SRVLOGIC-238: SonataFlow examples are missing quarkus 3 upgrade

### PR DESCRIPTION
**Pull request by nmirasch**
…rate runtime containers  (#1863)

* removed quay.io/kiegroup/kogito-runtime-jvm:latest by using the default the one for jdk17. Updated versions in readme files

* use data-index-postgresql nightly image

jira: https://issues.redhat.com/browse/SRVLOGIC-238

<details>
<summary>
How to replicate CI configuration locally?
</summary>

Build Chain tool does "simple" maven build(s), the builds are just Maven commands, but because the repositories relates and depends on each other and any change in API or class method could affect several of those repositories there is a need to use [build-chain tool](https://github.com/kiegroup/github-action-build-chain) to handle cross repository builds and be sure that we always use latest version of the code for each repository.
 
[build-chain tool](https://github.com/kiegroup/github-action-build-chain) is a build tool which can be used on command line locally or in Github Actions workflow(s), in case you need to change multiple repositories and send multiple dependent pull requests related with a change you can easily reproduce the same build by executing it on Github hosted environment or locally in your development environment. See [local execution](https://github.com/kiegroup/github-action-build-chain#local-execution) details to get more information about it.
</details>

<details>
<summary>
How to retest this PR or trigger a specific build:
</summary>

- for <b>pull request checks</b>  
  Please add comment: <b>Jenkins retest this</b>

- for a <b>specific pull request check</b>  
  Please add comment: <b>Jenkins (re)run [kogito-examples] tests</b>

- for <b>quarkus branch checks</b>  
  Run checks against Quarkus current used branch  
  Please add comment: <b>Jenkins run quarkus-branch</b>

- for a <b>quarkus branch specific check</b>  
  Run checks against Quarkus current used branch  
  Please add comment: <b>Jenkins (re)run [kogito-examples] quarkus-branch</b>

- for <b>quarkus main checks</b>  
  Run checks against Quarkus main branch  
  Please add comment: <b>Jenkins run quarkus-main</b>

- for a <b>specific quarkus main check</b>  
  Run checks against Quarkus main branch  
  Please add comment: <b>Jenkins (re)run [kogito-examples] quarkus-main</b>

- for <b>quarkus lts checks</b>  
  Run checks against Quarkus lts branch  
  Please add comment: <b>Jenkins run quarkus-lts</b>

- for a <b>specific quarkus lts check</b>  
  Run checks against Quarkus lts branch  
  Please add comment: <b>Jenkins (re)run [kogito-examples] quarkus-lts</b>
 
- for <b>native checks</b>  
  Run native checks  
  Please add comment: <b>Jenkins run native</b>

- for a <b>specific native check</b>  
  Run native checks 
  Please add comment: <b>Jenkins (re)run [kogito-examples] native</b>

- for <b>native lts checks</b>  
  Run native checks against quarkus lts branch
  Please add comment: <b>Jenkins run native-lts</b>

- for a <b>specific native lts check</b>  
  Run native checks against quarkus lts branch 
  Please add comment: <b>Jenkins (re)run [kogito-examples] native-lts</b>
 
</details>

<details>
<summary>
How to backport a pull request to a different branch?
</summary>

In order to automatically create a **backporting pull request** please add one or more labels having the following format `backport-<branch-name>`, where `<branch-name>` is the name of the branch where the pull request must be backported to (e.g., `backport-7.67.x` to backport the original PR to the `7.67.x` branch).

> **NOTE**: **backporting** is an action aiming to move a change (usually a commit) from a branch (usually the main one) to another one, which is generally referring to a still maintained release branch. Keeping it simple: it is about to move a specific change or a set of them from one branch to another.

Once the original pull request is successfully merged, the automated action will create one backporting pull request per each label (with the previous format) that has been added.

If something goes wrong, the author will be notified and at this point a manual backporting is needed.

> **NOTE**: this automated backporting is triggered whenever a pull request on `main` branch is labeled or closed, but both conditions must be satisfied to get the new PR created.
</details>

### CODE REVIEW

**Review by domhanak — APPROVED** [review]

File: serverless-workflow-examples/serverless-workflow-callback-quarkus/docker-compose/docker-compose.yml [hunk 1]
```diff
@@ -83,7 +83,7 @@ services:
 
   data-index:
     container_name: data-index
-    image: quay.io/kiegroup/kogito-data-index-postgresql:${KOGITO_VERSION}
+    image: quay.io/kiegroup/kogito-data-index-postgresql-nightly:latest
```
Comment (domhanak) [inline serverless-workflow-examples/serverless-workflow-callback-quarkus/docker-compose/docker-compose.yml:86]: @ricardozanini @nmirasch @rgdoliveira What should we put here? Latest does not make sense as it gets outdated. My suggestion is to pick one concrete version here as we used to do. So maybe `main-2024-02-09` ?

Comment (ricardozanini) [reply serverless-workflow-examples/serverless-workflow-callback-quarkus/docker-compose/docker-compose.yml:86]: I agree @domhanak!

**Review by rgdoliveira — APPROVED** [review]

File: serverless-workflow-examples/serverless-workflow-data-index-quarkus/README.md [hunk 2]
```diff
@@ -205,4 +207,4 @@ After a while (note that to you need give time for event to be consumed)  you sh
 
 ### Query process details in Data Index
 
-Data Index GraphQL UI is available at http://localhost:8080/q/graphql-ui/
\ No newline at end of file
+Data Index GraphQL UI is available at http://localhost:8180/graphiql/
```
Comment (domhanak) [inline serverless-workflow-examples/serverless-workflow-data-index-quarkus/README.md:210]: Is this correct? I am getting 404, what works is this one - `http://localhost:8180/q/graphql-ui/` I guess on midstream we miss a change?

**Review by ricardozanini — APPROVED** [review]

File: serverless-workflow-examples/serverless-workflow-data-index-quarkus/README.md (hunk 2 above)
Comment (nmirasch) [reply serverless-workflow-examples/serverless-workflow-data-index-quarkus/README.md:210]: I think http://localhost:8180/q/graphql-ui/ works in dev mode,  and the http://localhost:8180/graphiql/ is the endpoint when it's started with the provided docker-compose

**Comment by nmirasch** [comment]
@domhanak updated the other of nightly:latest to the fixed ..nightly:main-2024-02-09

**Comment by nmirasch** [comment]
@domhanak, @ricardozanini @rgdoliveira while testing the changes I realize another commit was missing, regarding the newsletter example. I've added the changes here 'Consolidate serverless workflow newsletter-subscription example' in order to be able to see it working after Q3 migration

**Review by ricardozanini — APPROVED** [review]

**Comment by domhanak** [comment]
`serverless-workflow-loanbroker-showcase`

Still contains openjdk11 references in docker files:
`/home/dhanak/git-workspace/kogito-examples/serverless-workflow-examples/serverless-workflow-loanbroker-showcase/aggregator|loanbroker-flow|loanbroker-ui/src/main/docker`

**Review by domhanak — CHANGES_REQUESTED** [review]
`serverless-workflow-loanbroker-showcase`

Still contains openjdk11 references in docker files:
`/home/dhanak/git-workspace/kogito-examples/serverless-workflow-examples/serverless-workflow-loanbroker-showcase/aggregator|loanbroker-flow|loanbroker-ui/src/main/docker/Dockerfile.jvm|Dockerfile.legacy-jar`

**Review by domhanak — APPROVED** [review]
Thanks @nmirasch 

One issue we found is - https://issues.redhat.com/browse/SRVLOGIC-244
but not a blocker. Let's merge.

**Pull request merged**